gmail/src/models/task.ts

```diff
@@ -0,0 +1,60 @@
+import { postJsonRpc } from "../utils/http";
+import { URLS } from "../const";
+import { State } from "../models/state";
+
+/**
+ * Represent a "project.task" record.
+ */
+export class Task {
+    id: number;
+    name: string;
+    projectName: string;
+
+    /**
+     * Unserialize the task object (reverse JSON.stringify).
+     */
+    static fromJson(values: any): Task {
+        const task = new Task();
+        task.id = values.id;
+        task.name = values.name;
+        task.projectName = values.projectName;
+        return task;
+    }
+
+    /**
+     * Parse the dictionary return by the Odoo endpoint.
+     */
+    static fromOdooResponse(values: any): Task {
+        const task = new Task();
+        task.id = values.task_id;
+        task.name = values.name;
+        task.projectName = values.project_name;
+        return task;
+    }
+
+    /**
+     * Make a RPC call to the Odoo database to create a task
+     * and return the ID of the newly created record.
+     */
+    static createTask(partnerId: number, projectId: number, emailBody: string, emailSubject: string): Task {
+        const url = State.odooServerUrl + URLS.CREATE_TASK;
+        const accessToken = State.accessToken;
+
+        const response = postJsonRpc(
+            url,
+            { email_body: emailBody, email_subject: emailSubject, partner_id: partnerId, project_id: projectId },
+            { Authorization: "Bearer " + accessToken },
+        );
+
+        const taskId = response ? response.task_id || null : null;
+
+        if (!taskId) {
+            return null;
+        }
+
+        return Task.fromJson({
```

Comment:
Actually we could do the same as for project, return task name in ``/mail_plugin/task/create`` and use it to enable any custom override people could do on naming, and be consistent with what is stored in DB .

Reply:
For the task we just return the ID, not the name
![screenshot_spect](https://user-images.githubusercontent.com/54316226/125922835-83b1a135-61c2-45c7-9bf5-a04d0b968003.png)

Reply:
Ok, I'll update the Odoo endpoint, so it's consistent with the project endpoint (currently `/mail_plugin/task/create` does not return the name :/)